tracing-journald/src/lib.rs

```diff
@@ -0,0 +1,111 @@
+use std::{fmt, io, io::Write, os::unix::net::UnixDatagram};
```

Comment:
I think everything that uses `UnixDatagram` has to be flagged with `cfg(unix)`; otherwise it will break our Windows tests on CI. 

I'm not totally sure if we want to make all the API types `cfg(unix)` so that they're not present at all on Windows, or if we want to feature flag the _internal_ implementation and just make them all nop (or have the constructor return an error?) on Windows. The latter would let Windows users still see the docs without having to pass a config flag, and would mean that code that uses the journald layer along with other layers wouldn't have to feature flag the use of the journald layer...what do you think makes the most sense?

Reply:
My knee-jerk reaction is that we should, at most, `cfg` out the entire crate on nonlinux (note that journald is in fact *Linux* specific, not just Unix). A noop implementation would take work to build/maintain, and cost buildtime for no sensible purpose. docs.rs already exposes platform-specific docs if a windows user really needs to see it for some reason.

Reply:
Though I guess we could leave it in on unix in general, since the worst that'll happen there is `new` failing?

Reply:
My thought process re: a nop implementation is just that if `new` fails because the user is on Windows, they can handle that error the same as how they would handle the error returned on a non-Linux Unix OS, or on a Linux distro without `systemd` (such as Void). I think it's a little easier to be able to handle _all_ fallbacks in the case where `new` fails, rather than having to have a separate "new exists but fails" case matching the return type and "all code in the crate is unavailable" case that's `cfg(not(unix))` or similar. 

This isn't a strongly held opinion, though, as long as there's a way to stop this from breaking the Windows CI, I'm fine with whatever you think is best.

Reply:
Had a go at this, it only took a few `#[cfg(unix)]`s so maybe not such a big maintenance issue after all.